website/addons/github/model.py

```diff
@@ -163,6 +170,20 @@ class AddonGitHubNodeSettings(AddonNodeSettingsBase):
 
     registration_data = fields.DictionaryField()
 
+    def find_or_create_file_guid(self, path):
+        try:
+            return GithubGuidFile.find_one(
+                Q('path', 'eq', path) &
+                Q('node', 'eq', self.owner)
+            ), False
+        except ModularOdmException:
```

Comment:
FWIW, this will have to be refactored anyway--I'll update #1601 and replace this logic once we're ready to get back to the issue of duplicate guids.